### PR TITLE
Longhorn: commit the code for flexvolume driver for longhorn

### PR DESCRIPTION
**Pull request by jimmy-peng**

### CODE REVIEW

File: driver/build/conf/longhorn [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+    if [ ! -d "${mntPath}" ]; then
+        echo -ne "${mntPath} does not exist"
+        return
+    fi
+
+    if ! OUT=$(mount "${fullDevName}" "${mntPath}" 2>&1); then
+        if [ $(echo ${OUT} | grep 'mount: wrong fs type,' | wc -l) -gt 0 ]; then
+            if ! OUT=$(mkfs -t ${fsType} ${fullDevName} >/dev/null 2>&1); then
+                echo -ne "Failed to create fs ${fsType} on device ${fullDevName}"
```
Comment (yasker) [inline driver/build/conf/longhorn]: Don't format the fs during mount. Do it when creating the device.

File: driver/build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM tutum/dnsutils
```
Comment (yasker) [inline driver/build/Dockerfile]: Can we use more standard base image? Like ubuntu. Tutum is Docker Container Cloud.

**Comment by jimmy-peng** [comment]
@yasker review again, thanks

**Comment by yasker** [comment]
I tried to test it again current master but failed. Also, where can I find the logs?

File: driver/build/conf/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+sed -ri "s|^LONGHORN_SVC=.*|LONGHORN_SVC=\"${LONGHORN_BACKEND_SVC_IP}:9500\"|" "/${DRIVER}"
+sed -ri "s|^LONGHORN_NODEID=.*|LONGHORN_NODEID=\"${NODE_NAME}\"|" "/${DRIVER}"
+
+driver_dir=$VENDOR${VENDOR:+"~"}${DRIVER}
+if [ ! -d "/flexmnt/$driver_dir" ]; then
+  mkdir "/flexmnt/$driver_dir"
+fi
+
+cp "/$DRIVER" "/flexmnt/$driver_dir/.$DRIVER"
+mv -f "/flexmnt/$driver_dir/.$DRIVER" "/flexmnt/$driver_dir/$DRIVER"
```
Comment (yasker) [inline driver/build/conf/entrypoint.sh]: The driver will be leftover on the host if the driver was deleted later. Need to find a way to uninstall it through deletion driver.

Comment (yasker) [reply driver/build/conf/entrypoint.sh]: Filled https://github.com/rancher/longhorn-manager/issues/14 for this.

**Pull request closed without merging**